src/journal-remote/journal-upload.c

```diff
@@ -208,6 +209,14 @@ int start_upload(Uploader *u,
                         return log_error_errno(SYNTHETIC_ERRNO(ENOSR),
                                                "Call to curl_easy_init failed.");
 
+                /* If configured, set a timeout for the curl operation */
+                if (arg_curl_timeout_secs > 0) {
+                        easy_setopt(curl, CURLOPT_TIMEOUT, arg_curl_timeout_secs,
+                                    LOG_ERR, return -EXFULL);
```

Comment:
convert the µs timeout i asked for above to secs as curl expects it via DIV_ROUND_UP() since we probably want to never timeout too early but rather too late